docs/modules/ROOT/pages/reference/framework-requirements.adoc

```diff
@@ -0,0 +1,46 @@
+= Framework Requirements
+
+As discussed on xref:explanations/what_is.adoc["What is it?"], a *framework* for self-service provisioning of services out of a Kubernetes cluster is a core part of the application catalog.
+
+This page discusses requirements the framework must adhere to:
+
+Reusability::
+Base parts of the framework should be reusable by the Open Source community and VSHNeers.
+When we develop a base part of the framework which might be usable by others, we try to bring it upstream or contribute it to the place it best belongs to.
+Just like Puppet base module or Helm charts are (can be) reusable.
+
+Low barrier to entry - xref:reference/glossary.adoc#_personas[Service Maintainer]::
+Adding a new service to the catalog using the framework should be as easy as possible.
+Required skills:
+Must:::
+* Kubernetes
+* Helm, Helm Charts
+* Prometheus, PromQL
+* (Crossplane)
+Should:::
+* Commodore Components
+
+Low barrier to entry - xref:reference/glossary.adoc#_personas[Service User]::
+Using a service should be a no-brainer:
+* Simple and discoverable interface
+* Expressive and approachable documentation
+
+Low maintenance::
+Our focus must be on maintaining the service itself.
+Core parts of the framework should be maintained by a community whenever possible or by the framework maintainer.
+(Just like Puppet itself is maintained by Puppet Labs - we maintain the modules and run the infrastructure)
+
+More buy than make::
+Try to use what's already available and focus on sticking parts together to be of use for us.
+Only engineer code (for example Go or Python) when it's for sticking parts together or a mandatory part is missing.
+
+Be aware of the risks::
+When "buying" a part of the system, do a risk assessment and document it properly.
+
+Testing facilities::
+Automated testing is important for the quality of a service.
+The framework must provide facilities for a service maintainer to be able to specify tests for a service.
+
+Identity::
+Services which are making use of the framework have a unique identity.
+To achieve that, the framework must help with that.
```

Comment:
I do not understand what is meant by this. In another paragraph it's required that the services have the same look, yet here we're talking about unique identity? What is meant with "identity"?

Reply:
I reshuffled the page a bit which should make this requirement more clear.

Reply:
Thanks. Though I still don't understand the "identity" bit. Is this some sort of unique name to avoid duplication?

Reply:
With identity, I mean a way to identify a service to be part of appcat.

Examples:
* All AppCat services share the same API group (e.g. `appcat.vshn.io/vX`)
* All AppCat services have a common XRD structure, so that common parameters between services are identical (e.g. `spec.sla`)

Reply:
Thanks for the explanation. I would never have guessed that the API group and consistency is meant with this :see_no_evil:  I guess this is confusing for others too. Can we maybe rephrase that to `Services which are making use of the framework share a common API scheme` or something along this? It's clearer than mentioning "identity" without explaining it somewhere.

Reply:
I agree. We don't have to specify it in detail yet, but the intention should be made clear.